protobuf/jaeger.proto

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package jaegertracing.protobuf;
+
+message Empty {};
+
+message ServiceRequest {
```

Comment:
can we keep this cleaner that thrift by separating data model from service definitions?

Reply:
Sure. I would really call this `common.proto`. `oneway` isn't supported in gRPC (https://github.com/grpc/grpc/issues/2147), so need a way to return nothing. Regarding `ServiceRequest`, Protobuf does not support request or response types that are not defined as `message` types beforehand, so no `rpc Request(string service_name) returns (Response)`. Had to come up with this to wrap the string in two services (see `baggage.proto` and `sampling.proto`).